source_pubsub/client.go

```diff
@@ -65,24 +66,37 @@ type PubSubClient struct {
 	codecCache  map[string]*goavro.Codec
 	unionFields map[string]map[string]struct{}
 
-	buffer chan sdk.Record
+	buffer chan ConnectResponseEvent
 	ticker *time.Ticker
```

Comment:
If each topic gets its own consumer, then each will need its own ticker. Since each ticker provides a channel and the channel is drained once per tick, it is unclear how many of the CDC consumers will actually execute a fetch.